source/ShipJumpNavigation.cpp

```diff
@@ -141,18 +142,38 @@ pair<JumpType, double> ShipJumpNavigation::GetCheapestJumpType(const System *fro
 	double hyperFuelNeeded = HyperdriveFuel();
 	// If these two systems are linked, or if the system we're jumping from has its own jump range,
 	// then use the cheapest jump drive available, which is mapped to a distance of 0.
+	const double distance = from->Position().Distance(to->Position());
 	double jumpFuelNeeded = JumpDriveFuel((linked || from->JumpRange())
-			? 0. : from->Position().Distance(to->Position()));
-	if(linked && hasHyperdrive && (!jumpFuelNeeded || hyperFuelNeeded <= jumpFuelNeeded))
+			? 0. : distance);
+	bool canJump = jumpFuelNeeded && (!from->JumpRange() || from->JumpRange() >= distance);
+	if(linked && hasHyperdrive && (!canJump || hyperFuelNeeded <= jumpFuelNeeded))
 		return make_pair(JumpType::HYPERDRIVE, hyperFuelNeeded);
-	else if(hasJumpDrive)
+	else if(hasJumpDrive && canJump)
```

Comment:
I don't think you're changes are necessary. JumpDriveFuel always returns the fuel needed for a jump drive that covers the distance passed, so checking the range again is not necessary.